crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -982,23 +982,41 @@ def reload_nova_services
       # online db migrations
       def run_online_migrations
         nova_controllers = ::Node.find("roles:nova-controller")
-        return if nova_controllers.empty?
+        heat_servers = ::Node.find("roles:heat-server")
+        return if nova_controllers.empty? && heat_servers.empty?
 
-        if nova_controllers.size > 1
-          nova_controllers.select! { |n| n[:fqdn] == n[:pacemaker][:founder] }
+        if nova_controllers.any?
+          if nova_controllers.size > 1
```

Comment:
maybe simpler like this?:
before:
```
return if nova_controllers.empty? && heat_server.empty?

if nova_controller.any?
  if nova_controller.size > 1
    nova_controllers.select! ...
  end
  node = nova_controllers.first
  ...
end
...
```

after:
```
nova_controllers.select! ... if nova_controllers.size > 1 # might result in long lines
heat_controllers.select! ... if heat_controllers.size > 1

# at this point *_controllers should contain non-ha node/founder or nothing
if nova_controllers.any?
  node = nova_controllers.first
  ...
end
...
```

Reply:
OK, good one